### ISSUE TITLE
Implement type of variable

### ISSUE THREAD

**Issue by carlos-montiers** [carlos-montiers/enhancedbatch#52]
I want add the type to our variables.
The idea is as next:
All the variables currently are of type string.
The idea is store the data using a structure like this:

- type: enum
- rawData: pointer
- stringData: pointer
- needsConversionToString: bool

Thus, the normal usage will have a type: string, and rawData and stringData will point to the same address, and needsConversionToString will have false.

With this we can create for example, other type of variables, for example: the binary type, useful for store png images or even better save the conversion to GpImage, in this case type: gpimage, the stringData can simple point to string: 'binary...' or we can use needsConversionToString for create a string representation (hex) for it, for only do this conversion when is requested the first time saving an unnecessary "dump".

This will also help us for store the internal structure of a keyboard pressed key, where all the information is stored internally as type: keyinfo.

Also can be useful for the calculator, for store decimal numbers in fraction notation. With that we can have the type: fraction. and do calculations without loss.

But I want primary for the need of have binary variables.

**Comment by lazna** [comment]
Not sure I this is suitable thread, but something like `field` will be usefull. Often using pseudo-fields: Sets of string variables numbered in its names:

```
var#1=aaa#bbb#ccc#ddd
var#2=eee#fff#ggg#hhh
etc..
```
I am sure it could be somehow significantly improoved....

**Comment by adoxa** [comment]
I don't think this is a suitable thread. Are you describing the variable, or its contents, or both? The variable sounds like an array, its value sounds like a split.

**Comment by lazna** [comment]
I do not know what you mean by split in this context. By array I mean what is described in [this article](https://www.tutorialspoint.com/batch_script/batch_script_arrays.htm), numbered set of line variables witch shared data structure (data_delimiter_data_delimiter_data). There are several problems with this technique, for example empty values on input which cause two separators togeather and later parsing problems, and many others. My general proposal it to increase this technique somehow or made some replacement. I known, not much specific proposal, but I am unable to say anything better :-/

**Comment by adoxa** [comment]
Oh, we might have only been discussing split in email; it would be something like:

```batch
set var=aaa#bbb#ccc#ddd
call @split split /v var #
:: now you have split1 = aaa, split2 = bbb, split3 = ccc, split4 = ddd and split# = 4.

That also demonstrates arrays, which simply append the index directly to the name, appending # for the count.

This is definitely a separate issue.
